test/ex_doc/formatter/html/autolink_test.exs

```diff
@@ -233,11 +233,41 @@ defmodule ExDoc.Formatter.HTML.AutolinkTest do
     assert Autolink.typespec(quote(do: (foo(1) :: bar | baz when bat: foo)), [], []) ==
            ~s[foo(1) :: bar | baz when bat: foo]
 
-    assert Autolink.typespec(quote(do: (really_long_name_that_will_trigger_multiple_line_breaks(1) :: bar | baz)), [], []) ==
-           ~s[really_long_name_that_will_trigger_multiple_line_breaks(1) ::\n  bar |\n  baz]
+    if formatter_available?() do
```

Comment:
another option to manage these tests is to have:

```elixir
test "add new lines on |" do
  ...
end

@tag :formatter
test "add new lines on | with formatter" do
  ...
end

@tag :no_formatter
test "add new lines on | without formatter" do
  ...
end
```

and then configure proper exclude in test_helper.exs

Reply:
I like the tag approach the most.